spec/lib/msf/core/rpc/v10/job_state_tracker_spec.rb

```diff
@@ -0,0 +1,159 @@
+require 'spec_helper'
+
+require 'msf/core/rpc/v10/job_state_tracker'
+
+RSpec.describe JobStateTracker do
+
+  context "With a 0.1 second ttl " do
```

Comment:
I'd be good to nuke this chunk of tests; ActiveSupport's MemoryStore is probably pretty well tested already, so there's not much need for us to duplicate that effort! 👀 

If you wanted to be sure that the `result_ttl` is passed correctly as a parameter to the memory store though, I believe you could do something like:

```ruby
before(:each) do
    allow(ActiveSupport::Cache::MemoryStore).to receive(:new).and_call_original

    JobStateTracker.new(result_ttl=0.1)
end

it { expect(ActiveSupport::Cache::MemoryStore).to have_received(:new).with(expires_in: 0.1) }
```

Reply:
yea I really wasn't happy with this, but I'm not strictly speaking testing the MemoryStore so much as I'm trying to test the jobtrackers ability to TTL something, if that makes sense? 
I suppose making sure the memory store gets the right value passed in _should_ do for that though, cheers